app-misc/anki/anki-24.04.1.ebuild

```diff
@@ -842,38 +841,48 @@ src_prepare() {
 		# Some parts of the runner build system expect to be in a git repository
 		mkdir .git || die
 
-		# Creating the pseudo venv early skips pip dependency checks in src_compile.
+		# Creating the pseudo venv early skips pip dependency checks in src_configure.
 		mkdir -p out/pyenv/bin || die
 		ln -s "${PYTHON}" out/pyenv/bin/python || die
 		# TODO: ln -s "${BROOT}/usr/bin/protoc-gen-mypy" out/pyenv/bin || die
+		if use doc; then
+			ln -s "${BROOT}"/usr/bin/sphinx-apidoc out/pyenv/bin || die
+			ln -s "${BROOT}"/usr/bin/sphinx-build out/pyenv/bin || die
+		fi
 
 		# Anki's Qt detection mechanism falls back to Qt5 Python bindings, if Qt6
 		# Python bindings don't get imported successfully.
 		if ! use qt6; then
 			sed -i "s/import PyQt6/raise ImportError/" qt/aqt/qt/__init__.py || die
 		fi
+
+		# Fix hardcoded runner location
+		export CARGO_TARGET_DIR="${S}"/out/rust
+		cbuild_dir="$(CHOST=${CBUILD:-${CHOST}} cargo_target_dir)"
+		sed "s,rust/release,${cbuild_dir##*out/}," \
+			-i build/ninja_gen/src/render.rs || die
+
+		# Separate src_configure from runner build
+		sed '/ConfigureBuild/d' -i build/ninja_gen/src/build.rs || die
+	fi
+}
+
+src_configure() {
+	cargo_src_configure
+	if use gui; then
+		CHOST=${CBUILD:-${CHOST}} cargo_src_compile -p configure
```

Comment:
Setting just `CHOST` is fine for `cargo_target_dir`, but I would replace this instance with `tc-env_build` from `toolchain-funcs`.

Reply:
But I still need to set `CHOST` so that `local -x CARGO_BUILD_TARGET=$(rust_abi)` picks up the `CBUILD` arch, right?

Reply:
I was going to say `tc-env_build` does that, but actually you're right, it doesn't! Perhaps it should, but go ahead and do both for now.

Reply:
Ok! Turns out I missed that some parts of the build system don't respect `CARGO_BUILD_TARGET`, causing the build to fail, which the recent changes to cargo.eclass (https://github.com/gentoo/gentoo/commit/27d469a2114b4ad0b3e682854c50c806753eb472) uncovered. I'll fix both issues in one go!

Reply:
@chewi I fixed the build now and included your suggestion. Sorry for the delay.. I've been busy these last couple of days and when I finally found the time to work on this PR, my PC for Gentoo development broke down.